### DISCUSSION TITLE
Help needed with dagster-airbyte

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17766] [Q&A]
Hi all,

I'm trying to use dagster-airbyte to manage Airbyte OSS but I've got the following issue :

```Request to Airbyte API failed: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /api/v1/connections/list (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0xffff95a5b400&gt;: Failed to establish a new connection: [Errno 111] Connection refused'))```
I think it's because I use the latest version of Airbyte API (<https://reference.airbyte.com/reference/start>) which is reachable via the port 8086.
For example, if I want to retrieve all the Airbyte connections, I can use `GET <http://localhost:8086/v1/connections>` instead of `POST <http://locahost:8000/api/v1/connections/list>`

Do you know if dagster-airbyte will be updated to use this new API ? Or please do you have another solution ?
Best regards :slightly_smiling_face:

Damien

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699351076495159?thread_ts=1699351076.495159&cid=C01U954MEER).*

**Comment by jamiedemaria** [comment]
How are you configuring your airbyte resource? The `AirbyteResource` has a `port` config option that you can set to customize the port

**Reply by amstramdam** [reply]
Hello, 

Here is my initial code to configure my Airbyte resource : 

```
from dagster import asset, Definitions
from dagster_airbyte import airbyte_resource, load_assets_from_airbyte_instance

airbyte_instance = airbyte_resource.configured(
    {
        "host": "localhost",
        "port": "8000",
        "username": "airbyte",
        "password": "password",
    }
)

airbyte_assets = load_assets_from_airbyte_instance(
    airbyte_instance,
    workspace_id="b30f7aad-3d91-482d-885b-490c4d4af8f7",
    key_prefix=["ab_"]
)

defs = Definitions(
    assets=[airbyte_assets],
    resources={},
    schedules=[],
)
```

In this configuration, the port was bad. So I tried like this : 

```
airbyte_instance = airbyte_resource.configured(
    {
        "host": "localhost",
        "port": "8006",
        "username": "airbyte",
        "password": "password",
    }
)
```

Here, the endpoint is `/api/v1/connections/list` instead of `/v1/connections` and the HTTP method is POST instead of GET.
Then, I got the following issue : 
```
dagster.builtin - ERROR - Request to Airbyte API failed: HTTPConnectionPool(host='localhost', port=8006): Max retries exceeded with url: /api/v1/connections/list (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xffffa933b400>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

Finally, I tried like this
```
from dagster_airbyte import AirbyteResource, load_assets_from_airbyte_instance

airbyte_instance = AirbyteResource(
    host="localhost",
    port="8006",
    # If using basic auth, include username and password:
    username="airbyte",
    password="password",
)
```
And like this 
```
from dagster_airbyte import AirbyteResource, load_assets_from_airbyte_instance

airbyte_instance = AirbyteResource(
    host="localhost",
    port="8000",
    # If using basic auth, include username and password:
    username="airbyte",
    password="password",
)
```

But I got the same errors. I tested the APIs with Postman and both of them work (cf. postman screenshots)

<img width="704" alt="Capture d’écran 2023-11-08 à 09 50 31" src="https://github.com/dagster-io/dagster/assets/15010251/cbfdab97-3346-4135-952e-37118c005fd5">
<img width="704" alt="Capture d’écran 2023-11-08 à 09 51 16" src="https://github.com/dagster-io/dagster/assets/15010251/e4b7a3d8-563f-4b44-99d2-74e87506b964">


After investigation, the first API (port 8006) is the Airbyte API : [https://reference.airbyte.com/reference/listworkspaces](https://reference.airbyte.com/reference/listworkspaces)
The second one (port 8000) is the configuration API  : [https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html)

But none of them work for me with dagster-airbyte : 
- Dagster version : 1.5.6
- Airbyte OSS version : 0.50.31

Do you have an idea ? Thanks for your help.

**Reply by jamiedemaria** [reply]
ah ok thanks for adding some more detail! @benpankow  i think this has reached the limits of my airbyte integration knowledge - can you take a look?

**Reply by amstramdam** [reply]
Hello @benpankow, did you have time to take a look ?

**Reply by rtesselli** [reply]
I'm following because I've the same issue

**Reply by benpankow** [reply]
Hello - might I ask how you're running Airbyte and how you are deploying Dagster? I am having trouble replicating this issue with Dagster `1.5.12` and Airbtre `0.50.36` running on Docker locally. The following assets load and sync correctly for me:

```python
airbyte_resource = AirbyteResource(
    username="airbyte", password="password", host="localhost", port=8000
)
airbyte_assets = load_assets_from_airbyte_instance(
    airbyte=airbyte_resource,
    workspace_id="714145f6-c081-4cb5-a551-efde2a530e0a",
    key_prefix=["ab_"],
)
```

**Comment by gera0220** [comment]
Are you deploying on Docker? I don't recall the exact error message at the moment, but if you are, my solution was to place Airbyte and Dagster within the same Docker network. Alternatively, a simpler option is to replace "localhost" with "host.docker.internal"

**Comment by sebastiancharrier** [comment]
Having the same Issue, I have Airbyte in a VM in GCP. 
It works from postman using the endpoint `8006/v1/connections/list` but Dagster is using `8006/api/v1/connections/get`

**Reply by sebastiancharrier** [reply]
Problem resolved, It actually works when I create the resource with the port 8000 instead 8006
